src/domain/payments/price-ratio.ts

```diff
@@ -15,20 +13,30 @@ export const PriceRatio = ({
     return new InvalidZeroAmountPriceRatioInputError()
   }
 
-  const convertFromUsd = (convert: UsdPaymentAmount): BtcPaymentAmount => {
-    const amountAsNumber =
-      (Number(convert.amount) * Number(btc.amount)) / Number(usd.amount)
-    const amount = Math.round(amountAsNumber)
-    const btcAmount = convert.amount === 0n ? 0n : amount === 0 ? 1n : amount
-    return paymentAmountFromSats(toSats(btcAmount))
+  const convertFromUsd = (
+    convert: UsdPaymentAmount,
+  ): BtcPaymentAmount | ValidationError => {
+    if (convert.amount === 0n) {
+      return paymentAmountFromNumber({ amount: 0, currency: WalletCurrency.Btc })
+    }
+
+    const amount = Math.round(
+      (Number(convert.amount) * Number(btc.amount)) / Number(usd.amount),
```

Comment:
Why is this math still using Number? We should stay in BigInt and multiply by the desired precision. Using `number` for anything to do with money is unreliable.